erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -1136,15 +1135,25 @@ def update_purchase_order_supplied_items(self):
 			select rm_item_code, reserve_warehouse
 			from `tabPurchase Order` po, `tabPurchase Order Item Supplied` poitemsup
 			where po.name = poitemsup.parent
-			and po.name = %s""", self.purchase_order))
+			and po.name = %s """, self.purchase_order))
+
+		transfered_qty = {}
 
 		#Update reserved sub contracted quantity in bin based on Supplied Item Details
 		for d in self.get("items"):
 			item_code = d.get('original_item') or d.get('item_code')
+			transfered_qty[item_code] = d.qty
 			reserve_warehouse = item_wh.get(item_code)
 			stock_bin = get_bin(item_code, reserve_warehouse)
 			stock_bin.update_reserved_qty_for_sub_contracting()
 
+		po = frappe.get_doc("Purchase Order", self.purchase_order)
+
+		for items in po.supplied_items:
+			items.transfered_qty = items.transfered_qty + transfered_qty.get(items.rm_item_code)
```

Comment:
What if user cancelled the stock entry, does transfered_qty will be reduced?

Reply:
In BOM, user may can add same item two times. In that case same item will be added multiple times in the "Purchase Order Item Supplied" table. As you have used the item code for the reference, the transfered_qty will show wrong for the 1st place item. For example: Item A has 2 qty in first row and Item A has 4 qty in second row. Now when user creates the stock entry, as per your design transferred qty for Item A at first row and second row will be 4 which is wrong.